### ISSUE TITLE
Update text/css registration

### ISSUE THREAD

**Issue by gsnedders** [w3c/csswg-drafts#5310]
Currently the text/css registration is that in [RFC2318](https://tools.ietf.org/html/rfc2318), which refers to CSS1, and its definition of the `charset` parameter seems completely bogus (and doesn't fulfil [RFC6657](https://tools.ietf.org/html/rfc6657)'s recommendations).

Rationally, I think this should be in the Snapshot, but as a WG Note I believe we can't have a normative registration there. So, um, CSS2 (which is the only place we reference RFC2318, though see #5299)? Syntax? Neither really makes sense IMO…

https://www.w3.org/2020/01/registering-mediatypes is the current W3C guidelines for registering a MIME type.

**Comment by svgeesus** [comment]
Yeah this should be updated. I'm not sure whether it needs an IETF WG to sponsor it. But also we don't need to make an RFC nowadays to register a new Internet Media Type.

I guess the first step is to use the current Internet Media Type registration template, and put that in some document. The Snapshot would be fine I think, it is just to get review before the IANA registration is updated.

**Comment by SebastianZ** [comment]
Updating the label to the latest snapshot. From skimming through https://www.w3.org/guide/editor/mediatypes.html#registration-process, it's not totally clear to me what's required here, but the process refers to the standard process for recommendations.

@gsnedders @svgeesus Maybe you can come up with something here?

Sebastian

**Comment by svgeesus** [comment]
I have done registrations before so sure, I can do it. Adding it to the snapshot makes sense as that is the current definition of CSS as a whole. I can then do the update with IANA once CSS 2026 is published.

**Comment by Crissov** [comment]
RFC 6838 section 4.11. suggests that media type registrations should describe how fragment identifiers are to be handled. 

I don’t think the [Linked Parameters Module](https://drafts.csswg.org/css-link-params-1/) (with `#param()`) will be stable enough to appear in CSS2026. Other proposals like #11509 (`@sheet`) have progressed even less far yet.

**Comment by svgeesus** [comment]
> Currently the text/css registration is that in [RFC2318](https://tools.ietf.org/html/rfc2318), which refers to CSS1, and its definition of the `charset` parameter seems completely bogus (and doesn't fulfil [RFC6657](https://tools.ietf.org/html/rfc6657)'s recommendations).

RFC 6657 has a[SHOULD](https://datatracker.ietf.org/doc/html/rfc6657#section-3) which says

a.  specify that the "charset" parameter is not used for the defined
       subtype, because the charset information is transported inside
       the payload (such as in "text/xml"), or

   b.  require explicit unconditional inclusion of the "charset"
       parameter, eliminating the need for a default value.

For CSS, to be web compatible, we can't _disallow_ `charset` (we do have `@charset` for internal labelling, but it is not mandatory and rarely used). We also can't _require_ `charset` because it is rarely used. And we can't declare a default charset because [the actual algorithm is more complicated](https://drafts.csswg.org/css-syntax/#determine-the-fallback-encoding)

So (as this is SHOULD not MUST, and Web Compat is a good reason) I believe this section should just link to CSS Syntax 3, TR version, [determine the fallback encoding of a stylesheet](https://www.w3.org/TR/css-syntax-3/#determine-the-fallback-encoding)

**Comment by svgeesus** [comment]
I'm also tempted to explicitly quote this part of CSS Syntax 3, for the guidance of those who won't follow the links:

> Stylesheet authors should author their stylesheets in UTF-8, and ensure that either an HTTP header (or equivalent method) declares the encoding of the stylesheet to be UTF-8, or that the referring document declares its encoding to be UTF-8.

**Comment by svgeesus** [comment]
For reference, [this is the `text/css` registration at the time of writing](https://www.iana.org/assignments/media-types/text/css)

```text
(per RFC2318)

MIME media type name: text

MIME subtype name: css

Required parameters: none

Optional parameters: charset

The syntax of CSS is expressed in US-ASCII, but a CSS file can
contain strings which may use any Unicode character. Any charset
that is a superset of US-ASCII may be used; US-ASCII, iso-8859-X
and utf-8 are recommended.

Encoding considerations:

For use with transports that are not 8-bit clean, quoted-
printable encoding is recommended since the majority of
characters will be CSS syntax and thus US-ASCII

Security considerations:

Applying a style sheet to a document may hide information
otherwise visible. For example, a very small font size may be
specified, or the display of certain document elements may be
turned off.

CSS style sheets consist of declarative property/value pairs
assigned to element selectors. They contain no executable code.

As with HTML documents, CSS style sheets may contain links to
other media (images, sounds, fonts, other style sheets) and those
links are typically followed automatically by software, resulting

in the transfer of files without the explicit request of the user
for each one. The security considerations of each linked file are
those of the individual registered types.

Interoperability considerations:

CSS has proven to be widely interoperable across computer
platforms, across Web browsers of different makes, and for import
and export in multiple authoring tools.

Published specification: see [1]

Applications which use this media type:

CSS is device-, platform- and vendor-neutral and is supported by
a wide range of Web user agents and authoring tools for
formatting HTML and XML documents.

Additional information:

Magic number(s): none
File extension(s): .css
Macintosh File Type Code(s): "css "
Object Identifier(s) or OID(s): none

Person & email address to contact for further information:

The authors of this memo.

Intended usage: COMMON

Author/Change controller:
```

**Comment by svgeesus** [comment]
Security considerations should say that CSS is a modular specification, and to consult the Security Considerations section of individual modules for details.

Published specification will be [`https://www.w3.org/TR/css/`](https://www.w3.org/TR/css/) which currently points to CSS 2025, will soon point to CSS 2026, and will point to the latest snapshot as the years go on.

I recall that we don't use Macintosh File Type Code(s) or Object Identifiers in new registrations.

The registration should say that it obsoletes  [RFC 2318](https://datatracker.ietf.org/doc/html/rfc2318)

Author/Change controller should be W3C.

I'm re-reading [RFC 6838](https://www.rfc-editor.org/rfc/rfc6838.html) to see what has changed or been added.

**Comment by svgeesus** [comment]
My first draft of an updated registration [is now in the CSS 2026 draft](https://drafts.csswg.org/css-2026/#media-type)

**Comment by svgeesus** [comment]
[RFC 6338 section 5.5 Change Procedures](https://www.rfc-editor.org/rfc/rfc6838.html#section-5.5)

**Comment by svgeesus** [comment]
[CSS 2026 published](https://www.w3.org/TR/css-2026/) with [updated `text/css` registration](https://www.w3.org/TR/css-2026/#media-type)

**Comment by svgeesus** [comment]
I [requested IANA review of the updated text/css media type](https://mailarchive.ietf.org/arch/msg/media-types/8mNJz6XmcKy_RV9xFMiJjluJAyo/)

**Comment by Crissov** [comment]
Hm, I’m now too late with that, but maybe the update registration should also have mentioned MUAs (i.e. email / Internet Mail clients) instead of just “Web browsers” and “Web user agents and authoring tools”.

**Comment by svgeesus** [comment]
The [updated media type is live](https://www.iana.org/assignments/media-types/text/css)
